scripts/onboarding/aks/onboarding-msi-bicep-syslog/existingClusterOnboarding.bicep

```diff
@@ -0,0 +1,110 @@
+@description('AKS Cluster Resource ID')
+param aksResourceId string
+
+@description('Location of the AKS resource e.g. "East US"')
+param aksResourceLocation string
+
+@description('Existing or new tags to use on AKS, ContainerInsights and DataCollectionRule Resources')
+param resourceTagValues object
+
+@description('Workspace Region for data collection rule')
+param workspaceRegion string
+
+@description('Full Resource ID of the log analitycs workspace that will be used for data destination. For example /subscriptions/00000000-0000-0000-0000-0000-00000000/resourceGroups/ResourceGroupName/providers/Microsoft.operationalinsights/workspaces/ws_xyz')
+param workspaceResourceId string
+
+@description('Array of allowed syslog levels')
+param syslogLevels array
+
+@description('Array of allowed syslog facilities')
+param syslogFacilities array
+
+var clusterSubscriptionId = split(aksResourceId, '/')[2]
+var clusterResourceGroup = split(aksResourceId, '/')[4]
+var clusterName = split(aksResourceId, '/')[8]
+var workspaceLocation = replace(workspaceRegion, ' ', '')
+var dcrNameFull = 'MSCI-${workspaceLocation}-${clusterName}'
+var dcrName = ((length(dcrNameFull) > 64) ? substring(dcrNameFull, 0, 64) : dcrNameFull)
+var associationName = 'ContainerInsightsExtension'
+var dataCollectionRuleId = resourceId(clusterSubscriptionId, clusterResourceGroup, 'Microsoft.Insights/dataCollectionRules', dcrName)
+
+resource aks_monitoring_msi_dcr 'Microsoft.Insights/dataCollectionRules@2022-06-01' = {
+  name: dcrName
+  location: workspaceRegion
```

Comment:
you should use workspaceLocation instead of workspaceRegion since the workspaceLocation trims the whitepspaces.